bridge/client/app/_services/data.service.ts

```diff
@@ -185,4 +194,21 @@ export class DataService {
         this._evaluationResults.next(event);
       });
   }
+
+  public sendApprovalEvent(approval: Trace, approve: boolean) {
+    this.apiService.sendApprovalEvent(approval, approve)
+      .pipe(take(1))
+      .subscribe(() => {
+        let root = this._projects.getValue().find(p => p.projectName == approval.data.project).services.find(s => s.serviceName == approval.data.service).roots.find(r => r.shkeptncontext == approval.shkeptncontext);
+        this.loadTraces(root);
+      });
+  }
+
+  private updateApprovals(root) {
+    if(root.traces.length > 0) {
+      this._openApprovals.next(this._openApprovals.getValue().filter(approval => root.traces.indexOf(approval) < 0));
+      if(root.traces[root.traces.length-1].type == EventTypes.APPROVAL_TRIGGERED)
+        this._openApprovals.next([...this._openApprovals.getValue(), root.traces[root.traces.length-1]].sort((a, b) => new Date(b.time).getTime() - new Date(a.time).getTime()));
```

Comment:
the date comparison is needed numerous times, maybe you can use a helper function (but not important now I would say)

Reply:
True. I create a follow-up issue on that #2011